core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMergeSink.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+import io.airlift.slice.Slice;
+import io.trino.spi.Page;
+
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+
+public interface ConnectorMergeSink
+{
+    /**
+     * Store the page(s) resulting from a merge.  The page consists of:
+     * <ul>
+     *     <li>Blocks 0..n-3 in page are the data column blocks.</li>
+     *     <li>Block n-2: Is the "operation" IntArrayBlock, whoe values are {@link MergeDetails#INSERT_OPERATION_NUMBER},
+     *         {@link MergeDetails#DELETE_OPERATION_NUMBER} or {@link MergeDetails#UPDATE_OPERATION_NUMBER}
+     *     </li>
+     *     <li>Block n-1 is a connector-specific rowId block, previously returned by
+     *         {@link ConnectorMetadata#getMergeRowIdColumnHandle(ConnectorSession, ConnectorTableHandle, MergeDetails)}
+     *     </li>
+     * </ul>
+     * @param page The page to store.
+     */
+    default void storeMergedRows(Page page)
+    {
+        throw new UnsupportedOperationException("This connector does not support row merge");
+    }
+
```

Comment:
the connector may need to do a _prolonged_ IO operation.
if it does it synchronously, the query may  be non-cancellable 


add `CompletableFuture<?> isBlocked()` and use it in `MergeOperator.isBlocked`

Reply:
As side conversations confirmed, this is correctly handled by the fact that `AbstractRowChangeOperator` and `ScanFilterAndProjectOperator` both talk to the pageSource.

Reply:
Can you point me where `AbstractRowChangeOperator` can return blocked?
i.e. It does not feel correct to push more updates into an updatable page source that declares "i am blocked", does it?

(also, if something required side conversations to confirm, it needs to be documented)